### PR TITLE
[TypeChecker] NFC: Add a test-case for rdar://85263844 regression

### PR DESCRIPTION
**Pull request by xedin**

### CODE REVIEW

**Comment by xedin** [comment]
@swift-ci please smoke test

File: test/Constraints/construction.swift [hunk 1]
```diff
@@ -264,3 +264,9 @@ func test_that_optionality_of_closure_result_is_preserved() {
     a.flatMap { (group: [S]) -> [S]? in s.map { group + [$0] } } // Ok
   })
 }
+
+// rdar://85263844 - initializer 'init(_:)' requires the types be equivalent
+func rdar85263844(arr: [(q: String, a: Int)]) -> AnySequence<(question: String, answer: Int)> {
+  AnySequence(arr.map { $0 }) // FIXME: This is supposed to type-check.
```
Comment (slavapestov) [inline test/Constraints/construction.swift:270]: One could argue it *shouldn't* type check also, I just wanted us to have the test case establishing the behavior one way or another :)

**Review by slavapestov — APPROVED** [review]
Thanks!

File: test/Constraints/construction.swift (hunk 1 above)
Comment (xedin) [reply test/Constraints/construction.swift:270]: Sure, but since it did, it should keep working at least in the major versions supported but the compiler…

**Comment by xedin** [comment]
@swift-ci please test Windows platforms

**Pull request merged**